find.go

```diff
@@ -475,16 +475,27 @@ func Nth[T any, N constraints.Integer](collection []T, nth N) (T, error) {
 
 // Sample returns a random item from collection.
 func Sample[T any](collection []T) T {
+	result := SampleBy(collection, rand.IntN)
+	return result
+}
+
+// SampleBy returns a random item from collection, using intFunc as the random index generator.
+func SampleBy[T any](collection []T, intFunc func(int) int) T {
```

Comment:
Isn't possible to define a type here?


```suggestion
// an interesting comment to be provided here
type randomSeedFunc func(int) int

func SampleBy[T any](collection []T, intFunc randomSeedFunc) T {
```

This way both SampleBy and [SamplesBy](https://github.com/samber/lo/pull/516#discussion_r1724931129) would use this common type.

And the purpose of the initFunc would be clearer

Reply:
I added a type `randomIntGenerator` instead.

Reply:
Thanks